packages/responsive/src/useMediaQueries.tsx

```diff
@@ -0,0 +1,169 @@
+    const updateMatches = () => {
+      const isSmallAfterUpdate = smallMatchMedia.matches || false
+      const isMediumAfterUpdate = mediumMatchMedia.matches || false
+      const isLargeAfterUpdate = largeMatchMedia.matches || false
+
+      setKaizenMatches({
+        isSmall: isSmallAfterUpdate,
+        isMedium: isMediumAfterUpdate,
+        isLarge: isLargeAfterUpdate,
+        isMediumOrSmaller: !isLargeAfterUpdate,
+        isMediumOrLarger: !isSmallAfterUpdate,
+      })
+    }
+
+    mediumMatchMedia.addEventListener("change", updateMatches, true)
+
+    return () => {
+      mediumMatchMedia.removeEventListener("change", updateMatches)
+    }
+  }, [])
+
+  // ---------------------------------------
+  // Create custom queries matches for initial state
+  // ---------------------------------------
+  const customQueryMatchMedias = new Map()
+  const customQueryMatches = {}
+  new Map(Object.entries(propQueries)).forEach((queryString, queryName) => {
+    const matchMedia = window.matchMedia(queryString)
+    customQueryMatches[queryName] = matchMedia.matches || false
+
+    // Store this matchMedia so that we can loop over it later and add event listeners
+    customQueryMatchMedias.set(queryName, matchMedia)
+  })
+
+  const [customMatches, setCustomMatches] = useState<{
+    [key: string]: boolean
+  }>(customQueryMatches)
+
+  // ---------------------------------------
+  // Create an event listener for each custom query
+  // ---------------------------------------
+  useEffect(() => {
+    const updateCustomMatches = (
+      matchMedia: MediaQueryList,
+      queryName: string
+    ) =>
+      setCustomMatches({
+        ...customQueryMatches,
+        [queryName]: matchMedia.matches || false,
+      })
+
+    const allEventListeners = new Map()
+    customQueryMatchMedias.forEach((matchMedia, queryName) => {
+      // Store this method so that we can reference it in the cleanup function below
+      const updateCustomMatchesForQuery = () => {
+        updateCustomMatches(matchMedia, queryName)
+      }
+      allEventListeners.set(matchMedia, updateCustomMatchesForQuery)
+
+      // Add the event listener
+      matchMedia.addEventListener("change", updateCustomMatchesForQuery, true)
+    })
+
+    return () => {
+      allEventListeners.forEach((eventListener, matchMedia) => {
+        matchMedia.removeEventListener("change", eventListener)
+      })
+    }
+  }, [])
+
+  // ---------------------------------------
+  // Create Kaizen helper components
+  // ---------------------------------------
+  const kaizenComponents = {
+    SmallOnly: (props: any) => <>{kaizenMatches.isSmall && props.children}</>,
+    MediumOnly: (props: any) => <>{kaizenMatches.isMedium && props.children}</>,
+    LargeOnly: (props: any) => <>{kaizenMatches.isLarge && props.children}</>,
+    MediumOrSmaller: (props: any) => (
+      <>{kaizenMatches.isMediumOrSmaller && props.children}</>
+    ),
+    MediumOrLarger: (props: any) => (
```

Comment:
Look, maybe nitpicky so up to you, but the `any` here could be replaced with something like `{children: React.ReactNode}` - in which `children` is a required prop.